Engines/Wine/Engine/Versions/script.js

```diff
@@ -120,17 +119,21 @@ module.getAvailableVersions = function (wizard) {
     return versionsJson;
 }
 
-
 module.getLatestStableVersion = function (wizard, architecture) {
-    return getLatestVersion(wizard, "upstream-linux-" + architecture, /^\d+\.0(\.\d+)?$/);
+    return getLatestVersion(wizard, `${this._wineDistribution}-${this._winePackage}-${architecture}`, /^\d+\.0(\.\d+)?$/);
+}
+
+module.getLatestStableVersion = function (wizard, distribution, winePackage, architecture) {
```

Comment:
Yes, this is how it should be. Just remove the other `getLatestStableVersion` and update `getLatestDevelopmentVersion` and `getLatestStagingVersion`. Then, ensure that it's called everywhere with the correct parameters.